src/Azure.Functions.Cli/Actions/AzureActions/PublishFunctionAppAction.cs

```diff
@@ -225,6 +225,40 @@ public override async Task RunAsync()
                 }
             }
 
+            if (functionApp.IsLinux && !functionApp.IsDynamic && !string.IsNullOrEmpty(functionApp.LinuxFxVersion))
+            {
+                var allImages = Constants.WorkerRuntimeImages.Values.SelectMany(image => image).ToList();
+                if (string.IsNullOrEmpty(allImages.FirstOrDefault(image => functionApp.LinuxFxVersion.Contains(image))))
```

Comment:
use `IndexOf(image, StringComparison.OrdinalIgnoreCase) != -1)` instead to get case insensitive matching. You can also use `.Any(` instead of `FirstOrDefault` and `IsNullOrEmpty`